File: code/angular/src/app/model2/models/Model2Inputs.ts
export class Model2Inputs {
  startYear: number = 2025;
  ageAtStart: number = 30;
  endYear: number = 2065;
  amountAtStart: number = 100000;
  annualContribution = 12000;
  allocations: Model2Allocation[] = [
    new Model2Allocation('Growth', 0.5, 0.05),
    new Model2Allocation('MinimalRisk', 0.5, 0)];
  meanAnnualReturn = 0.05;
  targetAge = 67
}

export class Model2Allocation {
  constructor(public name: string,
    public proportion: number,
    public meanAnnualReturn: number) {
  }
}

File: code/dotnet/PredictorLibrary/Model2/Inputs/Model2Inputs.cs
namespace PredictorLibrary.Model2.Inputs;

public class Model2Inputs {
    public int StartYear { get; set; }
    public int AgeAtStart { get; set; }
    public int EndYear { get; set; }
    public decimal AmountAtStart { get; set; }
    public decimal AnnualContribution { get; set; }
    public int TargetAge { get; set; }
    
    public IList<Allocation> Allocations { get; set; } = new List<Allocation>();
}